itest/rust/src/basis_test.rs

```diff
@@ -0,0 +1,184 @@
+        EulerOrder::YXZ,
+        EulerOrder::ZXY,
+        EulerOrder::ZYX,
+    ];
+
+    let vectors_to_test: Vec<Vector3> = vec![
+        Vector3::new(0.0, 0.0, 0.0),
+        Vector3::new(0.5, 0.5, 0.5),
+        Vector3::new(-0.5, -0.5, -0.5),
+        Vector3::new(40.0, 40.0, 40.0),
+        Vector3::new(-40.0, -40.0, -40.0),
+        Vector3::new(0.0, 0.0, -90.0),
+        Vector3::new(0.0, -90.0, 0.0),
+        Vector3::new(-90.0, 0.0, 0.0),
+        Vector3::new(0.0, 0.0, 90.0),
+        Vector3::new(0.0, 90.0, 0.0),
+        Vector3::new(90.0, 0.0, 0.0),
+        Vector3::new(0.0, 0.0, -30.0),
+        Vector3::new(0.0, -30.0, 0.0),
+        Vector3::new(-30.0, 0.0, 0.0),
+        Vector3::new(0.0, 0.0, 30.0),
+        Vector3::new(0.0, 30.0, 0.0),
+        Vector3::new(30.0, 0.0, 0.0),
+        Vector3::new(0.5, 50.0, 20.0),
+        Vector3::new(-0.5, -50.0, -20.0),
+        Vector3::new(0.5, 0.0, 90.0),
+        Vector3::new(0.5, 0.0, -90.0),
+        Vector3::new(360.0, 360.0, 360.0),
+        Vector3::new(-360.0, -360.0, -360.0),
+        Vector3::new(-90.0, 60.0, -90.0),
+        Vector3::new(90.0, 60.0, -90.0),
+        Vector3::new(90.0, -60.0, -90.0),
+        Vector3::new(-90.0, -60.0, -90.0),
+        Vector3::new(-90.0, 60.0, 90.0),
+        Vector3::new(90.0, 60.0, 90.0),
+        Vector3::new(90.0, -60.0, 90.0),
+        Vector3::new(-90.0, -60.0, 90.0),
+        Vector3::new(60.0, 90.0, -40.0),
+        Vector3::new(60.0, -90.0, -40.0),
+        Vector3::new(-60.0, -90.0, -40.0),
+        Vector3::new(-60.0, 90.0, 40.0),
+        Vector3::new(60.0, 90.0, 40.0),
+        Vector3::new(60.0, -90.0, 40.0),
+        Vector3::new(-60.0, -90.0, 40.0),
+        Vector3::new(-90.0, 90.0, -90.0),
+        Vector3::new(90.0, 90.0, -90.0),
+        Vector3::new(90.0, -90.0, -90.0),
+        Vector3::new(-90.0, -90.0, -90.0),
+        Vector3::new(-90.0, 90.0, 90.0),
+        Vector3::new(90.0, 90.0, 90.0),
+        Vector3::new(90.0, -90.0, 90.0),
+        Vector3::new(20.0, 150.0, 30.0),
+        Vector3::new(20.0, -150.0, 30.0),
+        Vector3::new(-120.0, -150.0, 30.0),
+        Vector3::new(-120.0, -150.0, -130.0),
+        Vector3::new(120.0, -150.0, -130.0),
+        Vector3::new(120.0, 150.0, -130.0),
+        Vector3::new(120.0, 150.0, 130.0),
+    ];
+
+    for order in euler_order_to_test.into_iter() {
+        for vector in vectors_to_test.iter() {
+            let vector = deg_to_rad(*vector);
+            let rust_basis = Basis::from_euler(order, vector);
+            let godot_basis = InnerBasis::from_euler(vector, order as i64);
+            assert!(
+                (rust_basis).is_equal_approx(&godot_basis),
+                "got = {rust_basis}, expected = {godot_basis}"
+            );
+        }
+    }
+}
+
+#[itest]
+fn basis_equiv() {
+    use crate::assert_inner_equivalent;
+    assert_inner_equivalent!(InnerBasis, TEST_BASIS.inverse(), |a, b| {
+        Basis::is_equal_approx(&a, &b)
+    });
```

Comment:
Can we not pass `Basis::is_equal_approx` directly? Maybe due to the extra referencing?

If not, we should at least consider storing the closure in a variable rather than repeating it again and again.